common/theme_generator.py

```diff
@@ -56,32 +86,45 @@ def add_scoped_style(self, name, scope, **kwargs):
         new_style = STYLE_TEMPLATE.format(name=name, scope=scope, properties=properties)
         self.styles.append(ElementTree.XML(new_style))
 
-    def get_new_theme_path(self, name):
-        """
-        Save the transformed theme to disk and return the path to that theme,
-        relative to the Sublime packages directory.
-        """
-        if not os.path.exists(os.path.join(sublime.packages_path(), "User", "GitSavvy")):
-            os.makedirs(os.path.join(sublime.packages_path(), "User", "GitSavvy"))
-
-        path_in_packages = os.path.join("User",
-                                        "GitSavvy",
-                                        "GitSavvy.{}.hidden-tmTheme".format(name))
+    def write_new_theme(self, name):
+        full_path = os.path.join(sublime.packages_path(), self.get_theme_path(name))
 
-        full_path = os.path.join(sublime.packages_path(), path_in_packages)
-
-        with util.file.safe_open(full_path, "wb") as out_f:
+        with util.file.safe_open(full_path, "wb", buffering=0) as out_f:
             out_f.write(STYLES_HEADER.encode("utf-8"))
             out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
 
-        return path_in_packages
 
-    def apply_new_theme(self, name, target_view):
-        """
-        Apply the transformed theme to the specified target view.
-        """
-        path_in_packages = self.get_new_theme_path(name)
+class JSONThemeGenerator(ThemeGenerator):
+    """
+    A theme generator for the new syntax `.sublime-color-scheme`
+    """
 
-        # Sublime expects `/`-delimited paths, even in Windows.
-        theme_path = os.path.join("Packages", path_in_packages).replace("\\", "/")
-        target_view.settings().set("color_scheme", theme_path)
+    hidden_theme_extension = "hidden-color-scheme"
+
+    def __init__(self, original_color_scheme):
```

Comment:
Should we share most of this init method with `XMLThemeGenerator` since it is covering cases where the file was not found?

Reply:
👍